cli/npm/global.d.ts

```diff
@@ -0,0 +1,44 @@
+type Rule =
+  | string
+  | RegExp
+  | {
+      /** @internal */
+      _ruleBrand: never;
+    };
+
+interface BaseGrammar<Keys extends string> {
+  name: string;
+
+  extras($: Record<Keys, Rule>): Rule[];
+
+  externals($: Record<Keys, Rule>): Rule[];
```

Comment:
Checking https://github.com/tree-sitter/tree-sitter/blob/master/cli/src/generate/grammar-schema.json, only `rules` and `name` are required.